Reset RequestStore before running delayed jobs

NobodysNightmare opened the pull request:
## OpenProject work package

This resolves the issue reported in https://community.openproject.org/work_packages/21175
## Description

Citing from the comment I added:

>  Includable in a delayed job to reset the request store before each run of the delayed job.
>  This should be done, because application code expects the RequestStore to be invalidated
>  between multiple requests. For a delayed job, each job execution is the thing that comes
>  closest to the concept of a new request.

We realized that we need to do this, because the `Setting` class would not invalidate its cache key otherwise. Leading to delayed jobs running with the same settings (e.g. `host_name`) forever.

NobodysNightmare committed:
let delayed jobs reset the request store

This is especially neccessary since our Settings use the request store to determine, whether they need to invalidate the cache key.
If the request store was never reset, the settings in our delayed jobs would be cached at an old value forever (until the next restart).

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-reset-request-store-in-jobs.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-reset-request-store-in-jobs.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

ulferts commented:
@NobodysNightmare as a note because we talked about it:

Please evaluate using a more generic name for the prepended module so that later on, developers can add additional functionality to be run before/after each delayed job. On the other hand, as you are modifying the `perform` method, which is specific for delayed jobs, it should include `DelayedJob` in it's name. Proposals:
- `DelayedJobTraits`
- `DelayedJobPreparer`

The `DelayedJob` could also become a module of it's own.

You also wanted to adapt the specs accordingly.

NobodysNightmare committed:
generalize module included in all delayed jobs

now also applies to other activities one might want to execute before all delayed jobs
(as discussed with @ulferts)

NobodysNightmare commented:
@ulferts Changed it as we discussed...

oliverguenther commented:
Looks good from my side. I like the more abstract `BeforeDelayedJob`, if @ulferts has no objections, this can be merged.
